### PR TITLE
Initial upgrade of MSAL 1.2 in MSAL Angular

### PR DESCRIPTION
**Pull request by jasonnutter**
## 1.0.0-alpha.0

Initial upgrade to use `msal@1.2.0`.

* `msal@1.2.0` package is now a peer dependency, and must be installed alongside `@azure/msal-angular`.
* `MsalModule.forRoot` now takes two arguement.
    * The first argument is the configuration object, which is the [same `Configuration` object](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/src/Configuration.ts) you would pass to `msal`.
    * The second argument is a `MsalAngularConfiguration` object, containing the values for `consentScopes`, `popUp`, and `extraQueryParameters`.
    * See the [updated sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/918830f9750a43567f1dbc01bc6492481d267ed6/samples/MSALAngularDemoApp/src/app/app.module.ts#L41) for an example of how to pass these configuration objects.
* The `acquireToken` and `login` methods now take a single `AuthenticationParameters` object as parameters.
* `getUser()` is now `getAccount()`.
* Broadcast events now emit objects, instead of just strings.

### CODE REVIEW

File: lib/msal-angular/package.json [hunk 1]
```diff
@@ -24,9 +24,14 @@
     "node": ">=0.8.0"
```
Comment (jasonnutter) [inline lib/msal-angular/package.json]: TODO: Bump version to `1.0.0-alpha.0`

File: lib/msal-core/src/Configuration.ts [hunk 2]
```diff
@@ -93,6 +94,9 @@ export type FrameworkOptions = {
     isAngular?: boolean;
     unprotectedResources?: Array<string>;
     protectedResourceMap?: Map<string, Array<string>>;
+    consentScopes?: Array<string>;
+    popUp?: boolean;
+    extraQueryParameters?: StringDict;
```
Comment (jasonnutter) [inline lib/msal-core/src/Configuration.ts]: TODO: Just have Angular hold these values, don't pass to core.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 3]
```diff
@@ -1956,7 +1956,7 @@ export class UserAgentApplication {
      *
      * returns the logger handle
      */
-    protected getLogger() {
+    public getLogger(): Logger {
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts]: TODO: Don't require this change in core.

File: lib/msal-core/src/index.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ export { UserAgentApplication } from "./UserAgentApplication";
 export { Logger } from "./Logger";
 export { LogLevel } from "./Logger";
 export { Account } from "./Account";
-export { Constants } from "./utils/Constants";
+export { Constants, TemporaryCacheKeys } from "./utils/Constants";
```
Comment (jasonnutter) [inline lib/msal-core/src/index.ts]: TODO: This change is already done in Core, pull in latest dev.

File: lib/msal-angular/src/msal.interceptor.ts [hunk 5]
```diff
@@ -11,56 +11,48 @@ import 'rxjs/add/operator/mergeMap'
...
+                        this.auth.clearCacheForScope(accessToken);
+                        this.broadcastService.broadcast('msal:notAuthorized', err.message);
                     }
-                    var msalError = new MSALError(JSON.stringify(err), "", JSON.stringify(scopes));
-                    this.broadcastService.broadcast('msal:notAuthorized', msalError);
-                }
-            })); //calling next.handle means we are passing control to next interceptor in chain
-        }
+
+                    throw err;
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.interceptor.ts]: TODO: remove this?

File: samples/MSALAngularDemoApp/src/app/app.component.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import {BroadcastService} from "@azure/msal-angular";
 import { MsalService} from "@azure/msal-angular";
 import {ProductService} from "./product/product.service";
 import {Subscription} from "rxjs/Subscription";
+import { AuthError, AuthResponse } from '../../../../lib/msal-core/lib-commonjs';
```
Comment (DarylThayil) [inline samples/MSALAngularDemoApp/src/app/app.component.ts]: can this work with lerna and not reaching in?

**Review by DarylThayil — APPROVED** [review]
lgtm in terms of what this is trying to short term accomplish

File: samples/MSALAngularDemoApp/src/app/app.component.ts (hunk 6 above)
Comment (jasonnutter) [reply samples/MSALAngularDemoApp/src/app/app.component.ts]: Ah this was VS Code auto-import shenanigans, I'll fix.

File: lib/msal-angular/src/msal-guard.service.ts [hunk 7]
```diff
@@ -8,89 +8,46 @@ import {MSAL_CONFIG, MsalService} from "./msal.service";
...
-                            }, function (error) {
-                                return false;
-                            });
-                    }
-                    else {
-                        this.authService.loginRedirect(this.config.consentScopes, this.config.extraQueryParameters);
-                    }
-                }
+        if (!this.authService.getAccount()) {
+            if (this.msalConfig.framework.popUp) {
```
Comment (pkanher617) [inline lib/msal-angular/src/msal-guard.service.ts]: minor thing, but comments would be good here.

File: lib/msal-angular/src/msal.interceptor.ts (hunk 5 above)
Comment (pkanher617) [reply lib/msal-angular/src/msal.interceptor.ts]: If you remove this won't errors just get swallowed by this handler?

**Review by pkanher617 — APPROVED** [review]
One question and comment, otherwise looks good.

File: lib/msal-angular/src/msal.interceptor.ts (hunk 5 above)
Comment (jasonnutter) [reply lib/msal-angular/src/msal.interceptor.ts]: It will get broadcast as `msal:notAuthorized`.

Comment (jasonnutter) [reply lib/msal-angular/src/msal.interceptor.ts]: I'll gather feedback on this, but for will keep as-is.

File: lib/msal-angular/src/msal-guard.service.ts (hunk 7 above)
Comment (jasonnutter) [reply lib/msal-angular/src/msal-guard.service.ts]: Yeah, I will definitely comment this library more. Think I'll do that in a separate PR.

**Pull request merged**